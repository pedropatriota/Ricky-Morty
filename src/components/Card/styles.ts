import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	flex-direction: column;
	width: 18rem;
`;

export const ImagContainer = styled.div<{ image: string }>`
	background-image: url(${props => props.image});
	background-repeat: no-repeat;
	background-size: cover;
	width: 100%;
	max-width: 18rem;
	height: 18rem;
	border-radius: 1rem 1rem 0 0;
`;

export const InfoContainer = styled.div`
	padding: 0.5rem;
	background-color: ${props => props.theme.colors.el};
	border: 1px solid ${props => props.theme.colors.el};
	border-radius: 0 0 1rem 1rem;

	h2 {
		margin-bottom: 1rem;
		color: ${props => props.theme.colors.text};
	}
	p {
		margin-bottom: 0.4rem;
		color: ${props => props.theme.colors.text};
	}
`;
