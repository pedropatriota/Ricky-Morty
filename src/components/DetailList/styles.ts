import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	gap: 0.75rem;
	flex-direction: column;
	margin: 0.75rem 0;
`;
