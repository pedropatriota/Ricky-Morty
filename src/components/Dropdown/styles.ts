import styled from 'styled-components';
import Select from 'react-select';

export const ReactSelect = styled(Select)`
  .react-select__control {
    width: min(90vw, 12.5rem);
    background-color: ${(props) => props.theme.colors.el};
    border-color: ${(props) => props.theme.colors.border};
    box-shadow: 0 0 0 transparent;
    height: 3.375rem;

    &:hover {
      border-color: ${(props) => props.theme.colors.border};
    }
  }

  .react-select__single-value {
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__menu {
    width: 12.5rem;
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__option,
  .react-select__option--is-focused 
  {
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__dropdown-indicator,
  .react-select__indicator-separator {
    display: none;
  }
`;
