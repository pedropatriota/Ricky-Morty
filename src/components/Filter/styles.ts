import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	align-items: flex-end;
	justify-content: space-between;
	flex-wrap: wrap;
	gap: 0.8rem;

	@media (max-width: 500px) {
		flex-wrap: wrap;
		gap: 1rem;
	}
`;
