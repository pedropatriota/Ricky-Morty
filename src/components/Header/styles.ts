import styled from 'styled-components';

export const Header = styled.header`
	padding: 0px min(5vw, 5rem);
	background-color: ${({ theme }) => theme.colors.bg};
	box-shadow: 0px 10px 15px -3px rgba(0, 0, 0, 0.1);
	margin-bottom: 4rem;
`;

export const Container = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
	flex-wrap: wrap;

	img {
		width: 100%;
		height: auto;
		max-width: 10rem;
	}
`;
