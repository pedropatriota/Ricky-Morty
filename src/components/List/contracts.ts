import { FetchNextPageOptions, InfiniteQueryObserverResult } from 'react-query';

type keyProps = {
	id: number;
	name: string;
	species: string;
	status: string;
	gender: string;
	image: string;
};

export interface ListProps {
	allData: keyProps[];
	fetchNextPage: (
		options?: FetchNextPageOptions | undefined
	) => Promise<InfiniteQueryObserverResult<any, unknown>>;
	hasNextPage?: boolean;
}
