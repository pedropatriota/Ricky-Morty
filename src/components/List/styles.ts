import styled from 'styled-components';
import InfiniteScroll from 'react-infinite-scroller';

export const Scroller = styled(InfiniteScroll)`
	margin-top: 2.5rem;
	display: grid;
	grid-template-columns: repeat(auto-fit, minmax(18rem, 1fr));
	gap: 1rem;
	padding: 0px min(5vw, 5rem);
`;
