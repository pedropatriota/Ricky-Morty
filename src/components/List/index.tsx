import Character from '../Character';
import { Frown } from 'lucide-react';
import { ListProps } from './contracts';
import { Scroller, IconContainer } from './styles';

const List = ({ allData, fetchNextPage, hasNextPage }: ListProps) => {
	return (
		<Scroller
			loadMore={() => fetchNextPage()}
			hasMore={hasNextPage}
			data-testid="scroller"
		>
			{allData?.length > 0 ? (
				allData?.map(({ ...props }) => <Character key={props.id} {...props} />)
			) : (
				<IconContainer>
					<Frown width="10rem" height="10rem" data-testid="icon-emptyList" />
				</IconContainer>
			)}
		</Scroller>
	);
};

export default List;
