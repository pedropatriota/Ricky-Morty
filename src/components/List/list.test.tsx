import List from '.';
import { render, screen, within } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { light } from '../../styles/themes';
import { MemoryRouter } from 'react-router-dom';

describe('ListComponent', () => {
	const fetchNextPage = jest.fn();
	const props = [
		{
			id: 4,
			name: 'Beth Smith',
			status: 'Alive',
			species: 'Human',
			gender: 'Female',
			image: 'https://rickandmortyapi.com/api/character/avatar/4.jpeg'
		},
		{
			id: 37,
			name: 'Beth Sanchez',
			status: 'Alive',
			species: 'Human',
			gender: 'Female',
			image: 'https://rickandmortyapi.com/api/character/avatar/37.jpeg'
		},
		{
			id: 38,
			name: 'Beth Smith',
			status: 'Alive',
			species: 'Human',
			gender: 'Female',
			image: 'https://rickandmortyapi.com/api/character/avatar/38.jpeg'
		}
	];

	it('should render an SVG when data is empty', () => {
		render(<List allData={[]} fetchNextPage={fetchNextPage} hasNextPage={false} />);

		const svg = screen.getByTestId('icon-emptyList');

		expect(svg).toBeInTheDocument();
	});

	it('should render and map the data', async () => {
		const { debug, container } = render(
			<ThemeProvider theme={light}>
				<MemoryRouter>
					<List
						allData={[...props]}
						fetchNextPage={fetchNextPage}
						hasNextPage={false}
					/>
				</MemoryRouter>
			</ThemeProvider>
		);

		const scrolledContainer = screen.getByTestId('scroller');
		const names = screen.getAllByText(/beth/i);

		expect(scrolledContainer).toBeTruthy();
		expect(names).toBeTruthy();
	});

	it('should match the snapshot', () => {
		const { container } = render(
			<ThemeProvider theme={light}>
				<MemoryRouter>
					<List
						allData={[...props]}
						fetchNextPage={fetchNextPage}
						hasNextPage={false}
					/>
				</MemoryRouter>
			</ThemeProvider>
		);
		expect(container).toMatchSnapshot();
	});
});
