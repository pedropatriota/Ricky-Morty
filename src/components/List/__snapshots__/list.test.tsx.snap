// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent should match the snapshot 1`] = `
<div>
  <div
    class="sc-fqkvVR BweYT"
    data-testid="scroller"
  >
    <a
      href="/4"
      style="text-decoration: none; display: block;"
    >
      <div
        class="sc-aXZVg bNflSf"
        data-testid="card"
      >
        <div
          class="sc-gEvEer igRMls"
          data-testid="card-image"
        />
        <div
          class="sc-eqUAAy jAJozm"
        >
          <h2>
            Beth Smith
          </h2>
          <p>
            <strong>
              Status
              : 
            </strong>
             
            Alive
          </p>
          <p>
            <strong>
              Species
              : 
            </strong>
             
            Human
          </p>
          <p>
            <strong>
              Gender
              : 
            </strong>
             
            Female
          </p>
        </div>
      </div>
    </a>
    <a
      href="/37"
      style="text-decoration: none; display: block;"
    >
      <div
        class="sc-aXZVg bNflSf"
        data-testid="card"
      >
        <div
          class="sc-gEvEer hydhuE"
          data-testid="card-image"
        />
        <div
          class="sc-eqUAAy jAJozm"
        >
          <h2>
            Beth Sanchez
          </h2>
          <p>
            <strong>
              Status
              : 
            </strong>
             
            Alive
          </p>
          <p>
            <strong>
              Species
              : 
            </strong>
             
            Human
          </p>
          <p>
            <strong>
              Gender
              : 
            </strong>
             
            Female
          </p>
        </div>
      </div>
    </a>
    <a
      href="/38"
      style="text-decoration: none; display: block;"
    >
      <div
        class="sc-aXZVg bNflSf"
        data-testid="card"
      >
        <div
          class="sc-gEvEer hPXxUj"
          data-testid="card-image"
        />
        <div
          class="sc-eqUAAy jAJozm"
        >
          <h2>
            Beth Smith
          </h2>
          <p>
            <strong>
              Status
              : 
            </strong>
             
            Alive
          </p>
          <p>
            <strong>
              Species
              : 
            </strong>
             
            Human
          </p>
          <p>
            <strong>
              Gender
              : 
            </strong>
             
            Female
          </p>
        </div>
      </div>
    </a>
  </div>
</div>
`;
